packages/rest/src/router/metadata.ts

```diff
@@ -72,10 +72,65 @@ export function api(spec: ControllerSpec) {
       'The @api decorator can be applied to constructors only.',
     );
     const apiSpec = resolveControllerSpec(constructor, spec);
-    Reflector.defineMetadata(API_SPEC_KEY, apiSpec, constructor);
+    const key = constructor.prototype._server
+      ? `${API_SPEC_KEY}:${constructor.prototype._server}`
+      : API_SPEC_KEY;
+    Reflector.defineMetadata(key, apiSpec, constructor);
   };
 }
 
+/**
+ * A decorator to mark a controller as belonging to a specific server.
+ * ```ts
+ * // In your application...
+ * class MyApplication extends Application {
+ *   constructor() {
+ *     super();
+ *     this.server(RestServer, 'foo');
+ *   }
+ * }
+ *
+ * // In one of your controllers
+ * @server('foo')
+ * class MyController {
+ *  // ...
+ * }
+ * ```
+ * @param name The name of the server bound to your application context.
+ */
+export function server(name: string) {
+  return function(constructor: Constructor<any>) {
+    const endpoints = Reflector.getOwnMetadata(ENDPOINTS_KEY, constructor);
+    if (endpoints) {
+      rescopeMetadata(ENDPOINTS_KEY, name, constructor, endpoints);
+    }
+    const spec = getControllerSpec(constructor);
+    if (spec) {
+      rescopeMetadata(API_SPEC_KEY, name, constructor, spec);
+    }
+    constructor.prototype._server = name;
```

Comment:
Modifying prototype of application-defined class is not a good idea. If we have two controller classes extend from the same base, this code will change the base class and impact all subclasses.

Reply:
I might need to create a metadata key for each constructor that contains the name of its server (or an array of servers for multi-server support).